draco/analysis/delay.py

```diff
@@ -555,63 +556,91 @@ class DelaySpectrumEstimatorBase(task.SingleTask, random.RandomTask):
     )
     dataset = config.Property(proptype=str, default="vis")
     average_axis = config.Property(proptype=str)
-    complex_timedomain = config.Property(proptype=bool, default=False)
+    complex_timedomain = config.Property(proptype=bool, default=True)
     initial_amplitude = config.Property(proptype=float, default=10.0)
 
-    def setup(self, telescope: io.TelescopeConvertible):
-        """Set the telescope needed to generate Stokes I.
+    @overload
+    def process(self, dataset: FreqContainerType) -> containers.DelaySpectrum:
+        ...
 
-        Parameters
-        ----------
-        telescope : TransitTelescope
-            The telescope object to use
-        """
-        self.telescope = io.get_telescope(telescope)
+    @overload
+    def process(
+        self, datasets: list[FreqContainerType]
+    ) -> containers.DelayCrossSpectrum:
+        ...
 
-    def process(self, ss: FreqContainerType) -> containers.DelaySpectrum:
+    def process(self, datasets):
         """Estimate the delay spectrum.
 
         Parameters
         ----------
-        ss
-            Data to transform. Must have a frequency axis and one other axis to
-            average over.
+        datasets
+            Dataset(s) to transform. A single dataset will have its delay power spectrum
+            estimated, a list of datasets will have all a full cross power spectrum
+            estimation. Each must have a frequency axis and one other axis to average
+            over.
 
         Returns
         -------
-        dspec : DelaySpectrum
+        dspec
         """
-        ss.redistribute("freq")
-
-        if self.dataset not in ss.datasets:
-            raise ValueError(
-                f"Specified dataset to delay transform ({self.dataset}) not in "
-                f"container of type {type(ss)}."
+        if not self.complex_timedomain:
+            warnings.warn(
+                (
+                    "On reflection the real timedomain approach didn't make much sense,"
+                    " so it is deprecated to set complex_timedomain=False."
+                ),
+                DeprecationWarning,
             )
 
-        if (
-            self.average_axis not in ss.axes
-            or self.average_axis not in ss.datasets[self.dataset].attrs["axis"]
-        ):
-            raise ValueError(
-                f"Specified axis to average over ({self.average_axis}) not in "
-                f"container of type {type(ss)}."
+        if not isinstance(datasets, list):
+            datasets = [datasets]
+            cross = False
+        else:
+            # cross = len(datasets) > 1
```

Comment:
curious why this check is disabled - wouldn't we want to use the cross spectrum only if there's more than one dataset?